### PR TITLE
Bluetooth: Mesh: Fix send input_complete before public key

### PR DESCRIPTION
**Pull request by LingaoM**
According Mesh Profile Spec 5.4.2.4 Authentication, if device
use Input OOB Authentication method, should send input complete
pdu after local public key has been acked.

`bt_mesh_input_string` or `bt_mesh_input_number` directly send
`input_complete`, however does not check whether the pub key has
been sent.

Mesh Provisioning timeout set to `60` seconds, so even this
probability is extremely low, it does not mean that there is no
such probability.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

### CODE REVIEW

File: subsys/bluetooth/mesh/prov.c [hunk 1]
```diff
@@ -73,12 +73,14 @@
...
-	WAIT_STRING,           /* Waiting for string input from user */
-	NOTIFY_INPUT_COMPLETE, /* Notify that input has been completed. */
-	PROVISIONER,           /* The link was opened as provisioner */
+	WAIT_PUB_KEY,           /* Waiting for local PubKey to be generated */
+	LINK_ACTIVE,            /* Link has been opened */
+	WAIT_NUMBER,            /* Waiting for number input from user */
+	WAIT_STRING,            /* Waiting for string input from user */
+	NOTIFY_INPUT_COMPLETE,  /* Notify that input has been completed. */
+	PROVISIONER,            /* The link was opened as provisioner */
+	PUB_KEY_SEND,           /* Public key sended */
```
Comment (joerchan) [inline subsys/bluetooth/mesh/prov.c:82]:
```suggestion
	PUB_KEY_SENT,           /* Public key has been sent */
```

File: subsys/bluetooth/mesh/prov.c [hunk 2]
```diff
@@ -596,7 +598,11 @@ int bt_mesh_input_number(uint32_t num)
 
 	sys_put_be32(num, &link.auth[12]);
 
-	send_input_complete();
+	if (atomic_test_bit(link.flags, PUB_KEY_SEND)) {
```
Comment (joerchan) [inline subsys/bluetooth/mesh/prov.c:601]: Maybe put this into a helper function?

**Comment by jhedberg** [comment]
How will this work if OOB Public Key is used? It seems like you may need to set the `PUB_KEY_SENT` flag if we've determined that the provisioner gets our public key OOB instead of us sending it?

**Comment by LingaoM** [comment]
> How will this work if OOB Public Key is used? It seems like you may need to set the `PUB_KEY_SENT` flag if we've determined that the provisioner gets our public key OOB instead of us sending it?

https://github.com/zephyrproject-rtos/zephyr/blob/e01a4df3fb8d4052addf2cd8db110e7f61e09c3e/subsys/bluetooth/mesh/prov.c#L228-L230
Currently device not support OOB public key provisioning.

**Comment by jhedberg** [comment]
> Currently device not support OOB public key provisioning.

Right. I'm a bit surprised no-one has wanted to add that yet though 😄

**Review by jhedberg — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Comment by carlescufi** [comment]
@trond-snekvik could you review as well please?

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**